OpenRA.Mods.RA/Traits/Minelayer.cs

```diff
@@ -160,27 +169,31 @@ public IEnumerable<Order> Order(World world, CPos cell, int2 worldPixel, MouseIn
 
 				if (mi.Button == Game.Settings.Game.MouseButtonPreference.Action && underCursor == null)
 				{
-					minelayer.World.CancelInputMode();
-					yield return new Order("PlaceMinefield", minelayer, false) { TargetLocation = cell };
+					foreach (var minelayer in minelayers)
+					{
+						minelayer.World.CancelInputMode();
+						yield return new Order("PlaceMinefield", minelayer, false) { TargetLocation = cell };
+					}
 				}
 			}
 
 			public void Tick(World world)
 			{
-				if (!minelayer.IsInWorld || minelayer.IsDead)
+				minelayers.RemoveAll(minelayer => !minelayer.IsInWorld || minelayer.IsDead);
+				if (!minelayers.Any())
 					world.CancelInputMode();
 			}
 
 			CPos lastMousePos;
 			public IEnumerable<IRenderable> Render(WorldRenderer wr, World world) { yield break; }
 			public IEnumerable<IRenderable> RenderAfterWorld(WorldRenderer wr, World world)
 			{
-				if (!minelayer.IsInWorld)
+				if (!minelayers.Any())
 					yield break;
 
-				var movement = minelayer.Trait<IPositionable>();
+				var movement = minelayers.First().Trait<IPositionable>();
 				var minefield = GetMinefieldCells(minefieldStart, lastMousePos,
-					minelayer.Info.TraitInfo<MinelayerInfo>().MinefieldDepth);
+					minelayers.First().Info.TraitInfo<MinelayerInfo>().MinefieldDepth);
```

Comment:
This might give unexpected results if you have two different kinds of minelayers that have different values for `MinefieldDepth`. I don't think it's a big problem in the end, though.